compiler/modulegraphs.nim

```diff
@@ -74,6 +80,18 @@ type
     Docgen2JsonPass
     Docgen2Pass
 
+  SuggestFileSymbolDatabase* = object
+    lineInfo*: seq[TinyLineInfo]
+    sym*: seq[PSym]
+    caughtExceptions*: seq[seq[PType]]
+    caughtExceptionsSet*: seq[bool]
```

Comment:
For memory concerns should this be `caughtExceptionsSet*: IntSet`? Where the index of `caughtExceptions` is stored inside.

Reply:
Yes, same for isDecl. This will further reduce the memory used. However, I was planning to do this in a separate PR, after this one is merged, because this PR is already large enough.

Reply:
Well, since this PR is being ignored for such a long time, I've decided to not wait for it to be merged, but to go ahead and implement the IntSet optimization as well. Here it is: https://github.com/nim-lang/Nim/pull/23202/commits/5ff0f229a390f24837576ca6d8c28889c72c5b16